### ISSUE TITLE
Keyboard resize is slow in iOS

### ISSUE THREAD

**Issue by aeharding** [ionic-team/capacitor-plugins#1904]
### Plugin(s)

Keyboard

### Capacitor Version

```
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 5.5.1
  @capacitor/core: 5.5.1
  @capacitor/android: 5.5.1
  @capacitor/ios: 5.5.1

Installed Dependencies:

  @capacitor/cli: 5.2.3
  @capacitor/android: 5.2.3
  @capacitor/core: 5.2.3
  @capacitor/ios: 5.2.3
```

### Platform(s)

iOS

### Current Behavior

On iOS, the app is incorrectly resized after the keyboard animation finishes, not before.

On Android, web view is correctly resized before keyboard animation starts.

### Expected Behavior

Web view is resized before keyboard animation starts.

### Code Reproduction

https://github.com/aeharding/test-keyboard-slow

#### iOS incorrect behavior
https://github.com/ionic-team/capacitor-plugins/assets/2166114/888abf46-c036-4959-9738-49140f91848a

#### Android correct behavior

https://github.com/ionic-team/capacitor-plugins/assets/2166114/feaadf20-688f-40e8-8a1d-a1df67c329bd

### Additional Context

Original issue that got closed as stale: https://github.com/ionic-team/capacitor/issues/1540

**Comment by stephan-fischer** [comment]
@aeharding  Yes i use capacitor 4 and not 5 because since capacitor 5 the keyboard closing on ios is really slow. 
And on closing a white background is flickering on my dark themed app.

**Comment by Achder** [comment]
I'm having the same issue. I took a quick look at the code and I'm wondering if this line is the culprit:
https://github.com/ionic-team/capacitor-plugins/blob/d5483ce9f666f1a38c25a11e63899fd35701ca45/keyboard/ios/Plugin/Keyboard.m#L145

It adds a delay to keyboard resizing code.
The delay has of length of UIKeyboardAnimationDurationUserInfoKey + 0.2 seconds
The delay is smaller when hiding the keyboard which is probably why hiding is a bit smoother.

With the current code we can't apply this fix correctly:
https://github.com/ionic-team/capacitor/issues/1540#issuecomment-735221275

With the delay in place the transition animation will start late and therefore the content will resize late again.

**Comment by Achder** [comment]
@aeharding Did you by any chance find a workaround for this?

**Comment by Achder** [comment]
I fiddled around with the code and removed the delay I mentioned above. Without the delay the experience is much better.
I'd love to provide a PR to fix this but I don't have enough inside into why the delay is hardcoded in the first place. 
@dallastjames could comment on the issue?

**Comment by Reiserro** [comment]
anyone got a fix for this, removing the delay dont fix the problem for me. if anyone got he solution in my bio is my discord, maybe we could help each other out!

**Comment by Achder** [comment]
@Reiserro If you're still on v5 try my forked version:
https://www.npmjs.com/package/@meerpohl/keyboard

I will soon provide a v6 fork.

**Comment by Reiserro** [comment]
@Achder  thanks my G its working! 
![Thumbs up hamster ](https://github.com/ionic-team/capacitor-plugins/assets/48568613/314a3bc6-c747-4ee9-ab53-2aeb12af8660)

**Comment by mpmua** [comment]
Issue still present in Capacitor 6, makes for a poor user experience on iOS devices.

**Comment by sam-hu** [comment]
Looking for a solution to this as well, in the meantime I found a workaround.

Capacitor is compatible with Cordova plugins. I installed https://github.com/cjpearson/cordova-plugin-keyboard and uninstalled `@capacitor/keyboard`, then put this JS line in my app startup: 
```
window.Keyboard.shrinkView(true);
```

Now I don't have an issue with delayed resizing

**Comment by Vasile-Peste** [comment]
Is there any plan to address this issue?

**Comment by suathh** [comment]
We are experiencing this issue also. This is one of the critical feedback our customer base has given us. We can not offer them to use ionic because of this particular issue cause they think all of the app will be slow when they see a slow keyboard on iOS.

**Comment by ZbignevS** [comment]
Hello, issue is fixed here: https://github.com/ionic-team/capacitor-plugins/pull/2230

**Comment by hedinasr** [comment]
@ZbignevS how can I use your fix without waiting for it to be merged?

**Comment by aeharding** [comment]
Hi all. There are currently two workarounds while we are waiting for a fix to be merged. **Please upvote this issue so the capacitor team may prioritize it!**

# Option 1: DIY resizing with keyboard events

You can follow the steps in this comment. It works fine, I've been using it for a while:

https://github.com/ionic-team/capacitor/issues/1540#issuecomment-735221275

# Option 2: Patching `@capacitor/keyboard`

The other option is to patch the npm package. This is the better option, IMO.

## npm users

Follow [this guide](https://capawesome.io/blog/how-to-patch-a-capacitor-plugin/) and make the changes that are present in [this commit](https://github.com/ionic-team/capacitor-plugins/pull/2233/files).

## pnpm users

pnpm has patching built-in. Make sure you are running latest pnpm for best experience. [Follow the instructions here](https://pnpm.io/cli/patch) to apply the changes that are present in [this commit](https://github.com/ionic-team/capacitor-plugins/pull/2233/files).

**Comment by Aarbel** [comment]
@jcesarmobile thanks a lot for all you work ! Can you check this forum post too https://forum.ionicframework.com/t/possible-to-have-smooth-keyboard-slide-ins/234267 ?
Having a keyboard / viewport lack of animation is the last elements that makes the hybrid apps feel "not native".

**Comment by Aarbel** [comment]
> [@aeharding](https://github.com/aeharding) Thanks for the guide. Excited for this to be fixed in the core plugin...

Very excited too, the keyboard needs that animation since while !
